lib/jsdom/living/geometry/DOMRect-impl.js

```diff
@@ -0,0 +1,83 @@
+"use strict";
+const DOMRectReadOnlyImpl = require("./DOMRectReadOnly-impl").implementation;
+
+class DOMRectImpl extends DOMRectReadOnlyImpl {
```

Comment:
The handling of reflected elements in this file seems like a bit of a layering violation, and does not show up at all in the spec for https://drafts.fxtf.org/geometry-1/#domrect. Is this a spec omission? Or is there an alternate implementation strategy that's closer to the spec, but we're choosing this one? In either case I think we need a comment, and perhaps we should file a spec issue if it's the former.

Reply:
The spec was written in a way that requires observing attribute changes (when attributes get changed, set DOMRect component values accordingly) as well as a setter hook (when DOMRect gets changed reserialize attribute). So I'd argue that there is a layering violation, but only for the setter part. (See https://svgwg.org/svg2-draft/types.html#InterfaceSVGAnimatedRect.)

In terms of the getter part, I chose a different implementation strategy that's closer to how the other reflection classes work: only get the values when the user asked for it / called the getter. DOMRect is only used in conjunction with SVGAnimatedRect, so even though this approach doesn't scale in terms of complexity of the DOMRect class in case there are other ways of reflecting a DOMRect, it should be fine in practice.

I will add a comment documenting this, if you are okay with this approach.

Reply:
Can we file an issue to add a reserialization hook to the DOMRect spec, and point to it from here?

For lazily-getting the values instead of eagerly, I am in favor of your strategy, with appropriate documentation of the isomorphism.

Reply:
Done: https://github.com/w3c/fxtf-drafts/issues/396.